### PR TITLE
Added type definition tests

### PR DESCRIPTION
**Pull request by 9renpoto**
- Fixed type definitions

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
+        ],
+        "noImplicitAny": true,
```
Comment (andrewiggins) [inline tsconfig.json]: Could we simplify this line and the next couple to `strict: true`?

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        "noImplicitThis": true,
+        "strictNullChecks": true,
+        "strictFunctionTypes": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "jsx": "react",
+        "jsxFactory": "h",
+        "experimentalDecorators": true,
```
Comment (andrewiggins) [inline tsconfig.json]: Let's remove this line since this library doesn't use decorators

**Comment by 9renpoto** [comment]
Thank you for the review. fixed it :)

**Comment by 9renpoto** [comment]
rebased

**Pull request merged**